virtool/subtractions/api.py

```diff
@@ -343,54 +340,59 @@ async def job_remove(req: aiohttp.web.Request):
     raise HTTPNoContent
 
 
-@routes.get("/subtractions/{subtraction_id}/files/{filename}")
+@routes.view("/subtractions/{subtraction_id}/files/{filename}")
 @routes.jobs_api.get("/subtractions/{subtraction_id}/files/{filename}")
-async def download_subtraction_files(req: aiohttp.web.Request):
-    """
-    Download a Bowtie2 index file or a FASTA file for the given subtraction.
-
-    """
-    db = req.app["db"]
-    pg = req.app["pg"]
-    subtraction_id = req.match_info["subtraction_id"]
-    filename = req.match_info["filename"]
-
-    document = await db.subtraction.find_one(subtraction_id)
-
-    if document is None:
-        raise NotFound()
-
-    if filename not in FILES:
-        raise HTTPBadRequest(text="Unsupported subtraction file name")
-
-    async with AsyncSession(pg) as session:
-        result = (
-            await session.execute(
-                select(SubtractionFile).filter_by(
-                    subtraction=subtraction_id, name=filename
+class SubtractionFiles(PydanticView):
+    async def get(self) -> Union[r200[VirtoolSubtractionFile], r400, r404]:
```

Comment:
This endpoint actually is for download a binary file.

Is there anything in `aiohttp-pydantic` for dealing with this? If not, how is it document in OpenAPI.